Dim-P/netdata
# logs-management: Add option to submit logs to system journal

##### Summary
This PR adds the configuration option `# submit logs to system journal = no` to certain log sources, so that when enabled, logs from those sources will be submitted to the system journal once they are collected.

This option is _not_ mutually exclusive with the `db mode = full` option, so that collected logs can be both stored in the logs management DB and also be submitted to the system journal at the same time.

Also, some other minor changes:

1. There is some refactoring taking place in 

```
            do{
                switch(p_file_info->log_type){
                    case FLB_TAIL:
                    case FLB_WEB_LOG:
                    case FLB_SERIAL: 
            .......
            .......
             }
```
that was using an `if / else` statements before.

2. The `prio_level` configuration option is added to `flb_kmsg` type logs.

3. A bug is fixed where the `min bytes` configuration option for `flb_serial` type sources was not used.

##### Test Plan

1. Enable the `# submit logs to system journal = no` for each of the applicable log sources and observe if the collected logs are submitted to the system journal with their fields, as expected. The `functions` tab of the UI can be used for this purpose (`systemd-journal` function).
2. Change the `prio level` value for the `[kmsg Logs]` source and observe if logs with higher priority than the one set are dropped. 

##### Additional Information


<details> <summary>For users: How does this change affect me?</summary>
  
</details>
